
<!DOCTYPE html>
<html>
<head>
  
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">

    <title>Nanopore data &mdash; McGill SI 2018</title>

    <link rel="shortcut icon" href="../../../img/favicon.ico">
    <link rel="stylesheet" href="../../../css/alabaster.css" type="text/css">
    <link rel="stylesheet" href="../../../css/alabaster-overrides.css" type="text/css">

    
      <link href="../../../css/sepsis.css" rel="stylesheet">
    

    
      <script src="../../../search/require.js"></script>
    
      <script src="../../../search/search.js"></script>
    

    

    <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9">

    
  
</head>
<body>

  <div class="document">
    <div class="documentwrapper">
      <div class="bodywrapper">
        <div class="body" role="main">
          
            <h1 id="nanopore-tutorial">Nanopore tutorial</h1>
<p>We will be using the Galaxy server that you have been using this week so far so please open that up.</p>
<h2 id="data-we-will-use-download-from-dropbox">Data we will use (Download from dropbox):</h2>
<ul>
<li><code>EM_079517.fasta</code></li>
<li><code>illumina_assembly_graph.fastg</code></li>
<li><code>O55_combined_miniasm.gfa</code></li>
<li><code>sakai_prophages.fa</code></li>
</ul>
<h2 id="programs-we-will-use-off-galaxy">Programs we will use off galaxy:</h2>
<ul>
<li><a href="https://ics.hutton.ac.uk/tablet/">Tablet</a> </li>
<li><a href="http://rrwick.github.io/Bandage/">Bandage</a></li>
</ul>
<h4 id="first-off-we-are-going-to-start-by-using-some-ebola-data-to-look-at-mapping-nanopore-reads-to-a-reference-we-are-going-to-use-bwa-mem-that-has-long-read-capability">First off, we are going to start by using some Ebola data to look at mapping nanopore reads to a reference. We are going to use BWA mem that has long read capability.</h4>
<h1 id="step-1-fastq-dump">Step 1: Fastq-dump</h1>
<p>First we need to retrieve Ebola data from NCBI. </p>
<ul>
<li>We will use the <code>Get Data</code> tab on the left hand side panel of Galaxy, under <code>Tools</code>. </li>
<li>Click on <code>Download and Extract Reads in FASTA/Q</code></li>
<li>Options will now be visible in the middle panel of galaxy</li>
<li>The input type should be <code>SRR accession</code></li>
<li>In Accession enter <code>ERR1014225</code> (This is the accession number for nanopore reads of an Ebola sample from the West African outbreak)</li>
<li>Select <code>gzip compressed fastq</code> for the output format</li>
<li>Click <code>Execute</code></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%209.52.33%20AM.png" /></p>
<ul>
<li>The job will then appear in the <code>History</code> panel on the right hand side of the screen</li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2010.07.13%20AM.png" /></p>
<ul>
<li>The job will turn green when it has finished</li>
</ul>
<h1 id="step-2-bwa-mem">Step 2: BWA-MEM</h1>
<p>Next we will map our nanopore Ebola reads to a reference.</p>
<ul>
<li>We will use the <code>NGS:Mapping</code> tab on the left hand side panel of Galaxy, under <code>Tools</code>. </li>
<li>Click on <code>Map with BWA-MEM</code></li>
<li>Options will now be visible in the middle panel of galaxy</li>
<li>Select <code>Use a genome from history and build index</code></li>
<li>The reference sequence will be preloaded into your galaxy history: <code>EM_079517.fasta</code> (This is a finished Ebola reference genome for strain Zaire)</li>
<li>For Algorithm select <code>Auto. Let BWA decide the best algorithm to use</code></li>
<li>Select <code>Single</code> for type of reads</li>
<li>Select <code>ERR1014225 (fastq-dump)</code> for fastq dataset</li>
<li>Select <code>Do not set</code> for read groups information</li>
<li>Select <code>3. Nanopore 2D-reads mode (-x ont2d)</code> for analysis mode  </li>
<li>Click <code>Execute</code></li>
</ul>
<p><img alt="Screenshot" src="../screenshots/Screen%20Shot%202018-06-07%20at%2010.07.13%20AM.png" /></p>
<h3 id="when-it-has-finished-running-we-will-download-the-mapped-reads">When it has finished running, we will download the mapped reads.</h3>
<ul>
<li>Click on the finished job (Called <code>Map with BWA-MEM</code>etc.) in the <code>History</code> panel on the right hand side panel </li>
<li>You will see a save icon under the details of the job</li>
<li>Click on it and the <code>Download dataset</code> and <code>Download bam_index</code> options will appear</li>
<li>We need both so click on one after the other </li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2010.32.52%20AM.png" /></p>
<h1 id="step-3-viewing-the-mapping-in-tablet">Step 3: Viewing the mapping in Tablet</h1>
<ul>
<li>We will load the downloaded data into Tablet by opening the Tablet program and clicking on <code>Open Assembly</code> in the top left hand side of the screen</li>
<li>For <code>Primary assembly file or URL:</code> click on <code>Browse...</code> and find the downloaded Bam file (you need the file with the extension <code>.bam</code> not <code>.bai</code>)</li>
<li>For <code>Reference/consensus file or URL:</code> click on <code>Browse...</code> and find the downloaded (from dropbox) reference genome <code>EM_079517.fasta</code></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2010.42.02%20AM.png" /></p>
<ul>
<li>Click <code>Open</code> and then click on the Contig <code>EM_079517</code> on the left hand panel of the screen and you will see your mapping results ready to browse</li>
<li>Take a few minutes to explore the program and the results</li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2010.44.03%20AM.png" /></p>
<h4 id="what-can-you-tell-from-the-alignment-about-the-method-of-sequencing">What can you tell from the alignment about the method of sequencing?</h4>
<h4 id="does-the-data-look-different-to-illumina-data-that-you-might-have-seen">Does the data look different to Illumina data that you might have seen?</h4>
<h4 id="can-you-find-anything-that-looks-like-consistent-errors">Can you find anything that looks like consistent errors?</h4>
<h4 id="can-you-find-anything-that-looks-like-a-reliable-snp">Can you find anything that looks like a reliable SNP?</h4>
<h1 id="step-4-comparing-illumina-and-nanopore-assemblies">Step 4: Comparing Illumina and Nanopore assemblies</h1>
<h4 id="next-were-going-to-use-escherichia-coli-to-illustrate-how-much-assemblies-can-be-improved-with-long-reads">Next, we&rsquo;re going to use <em>Escherichia coli</em> to illustrate how much assemblies can be improved with long reads</h4>
<h2 id="pre-knowledge">Pre-knowledge</h2>
<ul>
<li>Pathogenic <em>E. coli</em> tends to have a lot repeat sequences</li>
<li>This is due to large prophages integrating in the genome</li>
<li>These prophages are very similar to each other and cause confusion for assembly algorithms </li>
<li>These prophages can be as large as ~50kb</li>
<li>No matter how deep you sequence, if you don&rsquo;t get reads longer than those repeats you will never finish the genome</li>
<li>See <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-9-r101">this paper</a> for more detail</li>
</ul>
<h2 id="bandage">Bandage</h2>
<div class="codehilite"><pre><span></span>Nodes = contigs
Edges = overlaps
</pre></div>


<ul>
<li>Open up Bandage from your desktop</li>
<li>Click on <code>File</code> then <code>Load graph</code></li>
<li>Find the <code>illumina_assembly_graph.fastg</code> location on your computer and click <code>Open</code></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2011.20.11%20AM.png" /></p>
<ul>
<li>Click <code>Draw graph</code> on the left hand panel of the screen</li>
<li>You will see that the Illumina assembly produces a whole mess of 649 contigs with a lot of short contigs connecting everything together</li>
<li>You can zoom in on the graph to look at it in more detail</li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2012.40.28%20PM.png" /></p>
<ul>
<li>We can colour the graph by depth by changing <code>Graph display</code> from <code>Random colours</code> to <code>Colour by depth</code></li>
<li>You&rsquo;ll see that the short contigs all have much higher coverage than the other contigs. <em>What do you think that means?</em></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2012.47.48%20PM.png" /></p>
<h4 id="now-we-are-going-to-use-blast-to-look-for-the-prophage-regions-in-the-assembly">Now we are going to use <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> to look for the prophage regions in the assembly</h4>
<ul>
<li>Click on <code>Create/view BLAST search</code> in the left hand panel of the screen</li>
<li><code>Step 1</code> click on <code>Build BLAST database</code></li>
<li><code>Step 2</code> click on <code>Load from FASTA file</code> and find the <code>sakai_prophages.fa</code> location on your computer</li>
<li><code>Step 3</code> click on <code>Run BLAST search</code></li>
<li>click on <code>Close</code></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2012.52.56%20PM.png" /></p>
<h4 id="you-will-see-the-prophage-regions-highlighted-in-colour-on-the-assembly-graph-now">You will see the prophage regions highlighted in colour on the assembly graph now.</h4>
<h4 id="what-affect-are-the-prophage-regions-having-on-the-assembly">What affect are the prophage regions having on the assembly?</h4>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2012.54.03%20PM.png" /></p>
<h4 id="now-we-will-load-the-nanopore-assembly-into-bandage-to-see-how-long-reads-affect-this">Now, we will load the nanopore assembly into Bandage to see how long reads affect this</h4>
<ul>
<li>Click on <code>File</code> then <code>Load graph</code></li>
<li>Find the <code>O55_combined_miniasm.gfa</code> location on your computer and click <code>Open</code></li>
<li>Click <code>Draw graph</code> on the left hand panel of the screen</li>
<li>Change <code>Graph display</code> option from <code>BLAST hits(solid)</code> to <code>Random colours</code>. <em>What is strikingly different about this assembly than the Illumina one?</em></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2012.58.17%20PM.png" /></p>
<h3 id="now-lets-look-at-where-the-prophage-regions-are-in-this-assembly">Now, lets look at where the prophage regions are in this assembly</h3>
<ul>
<li>Click on <code>Create/view BLAST search</code> in the left hand panel of the screen</li>
<li><code>Step 1</code> click on <code>Build BLAST database</code></li>
<li><code>Step 2</code> click on <code>Load from FASTA file</code> and find the <code>sakai_prophages.fa</code> location on your computer</li>
<li><code>Step 3</code> click on <code>Run BLAST search</code></li>
<li>click on <code>Close</code></li>
</ul>
<p><img alt="Alt text" src="../screenshots/Screen%20Shot%202018-06-07%20at%2012.59.42%20PM.png" /></p>
<h4 id="how-have-the-long-reads-improved-this-assembly">How have the long reads improved this assembly?</h4>
<h4 id="are-there-any-regions-that-havent-been-resolved-by-the-nanopore-experiment-why-would-that-be">Are there any regions that haven&rsquo;t been resolved by the nanopore experiment? Why would that be?</h4>
            
          
        </div>
      </div>
    </div>
    <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
      <div class="sphinxsidebarwrapper">
        
  <h1 class="logo"><a href="../../..">McGill SI 2018</a></h1>



  <p class="blurb">Tutorials for analysing bacterial omics data</p>

        
          



<h3>Table Of Contents</h3>

<nav>
  
  
    <ul>
    
      <li><a href="#data-we-will-use-download-from-dropbox">Data we will use (Download from dropbox):</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#programs-we-will-use-off-galaxy">Programs we will use off galaxy:</a></li>
      <ul>
    
      <li><a href="#first-off-we-are-going-to-start-by-using-some-ebola-data-to-look-at-mapping-nanopore-reads-to-a-reference-we-are-going-to-use-bwa-mem-that-has-long-read-capability">First off, we are going to start by using some Ebola data to look at mapping nanopore reads to a reference. We are going to use BWA mem that has long read capability.</a></li>
      <ul>
    
  </ul>
    
  </ul>
    
  </ul>
  
    <ul>
    
  </ul>
  
    <ul>
    
      <li><a href="#when-it-has-finished-running-we-will-download-the-mapped-reads">When it has finished running, we will download the mapped reads.</a></li>
      <ul>
    
  </ul>
    
  </ul>
  
    <ul>
    
      <li><a href="#what-can-you-tell-from-the-alignment-about-the-method-of-sequencing">What can you tell from the alignment about the method of sequencing?</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#does-the-data-look-different-to-illumina-data-that-you-might-have-seen">Does the data look different to Illumina data that you might have seen?</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#can-you-find-anything-that-looks-like-consistent-errors">Can you find anything that looks like consistent errors?</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#can-you-find-anything-that-looks-like-a-reliable-snp">Can you find anything that looks like a reliable SNP?</a></li>
      <ul>
    
  </ul>
    
  </ul>
  
    <ul>
    
      <li><a href="#next-were-going-to-use-escherichia-coli-to-illustrate-how-much-assemblies-can-be-improved-with-long-reads">Next, we're going to use Escherichia coli to illustrate how much assemblies can be improved with long reads</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#pre-knowledge">Pre-knowledge</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#bandage">Bandage</a></li>
      <ul>
    
      <li><a href="#now-we-are-going-to-use-blast-to-look-for-the-prophage-regions-in-the-assembly">Now we are going to use BLAST to look for the prophage regions in the assembly</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#you-will-see-the-prophage-regions-highlighted-in-colour-on-the-assembly-graph-now">You will see the prophage regions highlighted in colour on the assembly graph now.</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#what-affect-are-the-prophage-regions-having-on-the-assembly">What affect are the prophage regions having on the assembly?</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#now-we-will-load-the-nanopore-assembly-into-bandage-to-see-how-long-reads-affect-this">Now, we will load the nanopore assembly into Bandage to see how long reads affect this</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#now-lets-look-at-where-the-prophage-regions-are-in-this-assembly">Now, lets look at where the prophage regions are in this assembly</a></li>
      <ul>
    
      <li><a href="#how-have-the-long-reads-improved-this-assembly">How have the long reads improved this assembly?</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#are-there-any-regions-that-havent-been-resolved-by-the-nanopore-experiment-why-would-that-be">Are there any regions that haven't been resolved by the nanopore experiment? Why would that be?</a></li>
      <ul>
    
  </ul>
    
  </ul>
    
  </ul>
    
  </ul>
  

  
</nav>
          
  <h3>Related Topics</h3>
  <ul>
    
      <li>Previous: <a href="../../abricate/" title="previous chapter">
        Finding antibiotic-resistance genes
      </a></li>
    
    
      <li>Next: <a href="../../kraken/" title="next chapter">
        Sample identification with Kraken
      </a></li>
    
  </ul>

        
        <div id="searchbox" style="display: none;" role="search">
  <h3>Quick search</h3>
  <form class="search" action="../../../search.html" method="get">
    <input name="q" type="text">
    <input value="Go" type="submit">
  </form>
  <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
  </p>
</div>
<script type="text/javascript">
  document.getElementById("searchbox").style.display = "block";
</script>
      </div>
    </div>
    <div class="clearer"></div>
  </div>

  
    <div class="footer">
      
        &copy; &copy; The authors; 2016-
      
      
        |
        Powered by <a href="http://www.mkdocs.org">MkDocs 0.17.2</a>
        &amp; <a href="https://github.com/iamale/mkdocs-alabaster">Alabaster</a>
      
    </div>
  

  <!--
  MkDocs version      : 0.17.2
  Docs Build Date UTC : 2018-06-13 05:08:11
  -->
</body>
</html>