
<!DOCTYPE html>
<html>
<head>
  
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">

    <title>Quality control &mdash; McGill SI 2018</title>

    <link rel="shortcut icon" href="../../img/favicon.ico">
    <link rel="stylesheet" href="../../css/alabaster.css" type="text/css">
    <link rel="stylesheet" href="../../css/alabaster-overrides.css" type="text/css">

    
      <link href="../../css/sepsis.css" rel="stylesheet">
    

    
      <script src="../../search/require.js"></script>
    
      <script src="../../search/search.js"></script>
    

    

    <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9">

    
  
</head>
<body>

  <div class="document">
    <div class="documentwrapper">
      <div class="bodywrapper">
        <div class="body" role="main">
          
            <p><br></p>
<h1 id="fastqc-in-galaxy">FastQC in Galaxy</h1>
<!---
FIXME: include screenshots
FIXME: include file location and/or choose different input.
--->

<p>After sequencing, the reads should be checked for their quality. This tutorial demonstrates how to use the tool called FastQC to examine bacterial paired-end Illumina sequence reads. The FastQC website is <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here.</a></p>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>run FastQC on input sequence reads, and</li>
<li>examine the FastQC output.</li>
</ol>
<h2 id="input-files">Input files</h2>
<p>See <a href="/modules/galaxy/index.md">here</a> for information about how to start with Galaxy, and <a href="../data-dna/">here</a> for the link to import the &ldquo;Galaxy history of input files&rdquo; for this tutorial, if you don&rsquo;t already have them in your history.</p>
<ul>
<li>We will use the paired-end read set from our Galaxy history.</li>
<li>These are the files <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>.</li>
<li>FASTQ is a file format for sequence reads that displays quality scores for each of the sequenced nucleotides. For more information about FASTQ format see this <a href="https://en.wikipedia.org/wiki/FASTQ_format">link</a>.</li>
<li>We will evaluate the <fn>mutant_R1.fastq</fn> reads using the FastQC tool.</li>
</ul>
<h2 id="run-fastqc">Run FastQC</h2>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: QC and Manipulation &rarr; FastQC</ss></li>
</ul>
<p><img alt="FastQC selection" src="./images/image04.png" /></p>
<ul>
<li>for <ss>Short read data from your current history</ss>: <fn>mutant_R1.fastq</fn></li>
<li>Click <ss>Execute</ss></li>
<li>In the History pane, click on the &ldquo;refresh&rdquo; icon to see if the analysis has finished.</li>
</ul>
<h2 id="examine-output-files">Examine output files</h2>
<p>Once finished, examine the output called <fn>FastQC on data1:webpage</fn> (Hint: click the eye icon). It has a summary at the top of the page and a number of graphs.</p>
<p>Look at:</p>
<ul>
<li>
<p><ss>Basic Statistics</ss></p>
<ul>
<li><ss>Sequence length</ss>: will be important in setting maximum k-mer size value for assembly.</li>
<li><ss>Encoding</ss>: The quality encoding type is important for quality trimming software.</li>
<li><ss>% GC</ss>: high GC organisms donâ€™t tend to assemble well and may have an uneven read coverage distribution.</li>
<li><ss>Total sequences</ss>: Total number of reads: gives you an idea of coverage.</li>
</ul>
</li>
<li>
<p><ss>Per base sequence quality</ss>: Dips in quality near the beginning, middle or end of the reads: determines possible trimming/cleanup methods and parameters and may indicate technical problems with the sequencing process/machine run. In this case, all the reads are of relatively high quality across their length (150 bp).</p>
</li>
</ul>
<p><img alt="sequence quality graph" src="./images/seq_quality.png" /></p>
<ul>
<li>
<p><ss>Per base N content</ss>: Presence of large numbers of Ns in reads may point to a poor quality sequencing run. You would need to trim these reads to remove Ns.</p>
</li>
<li>
<p><ss>Kmer content</ss>: Presence of highly recurring k-mers: may point to contamination of reads with barcodes, adapter sequences etc. In this case, we have spikes in two types of kmers. <!-- explain why?
--></p>
</li>
</ul>
<p><img alt="kmer content graph" src="./images/kmer_content.png" /></p>
<p>We have warnings for two outputs (per base sequence content; Kmer content). This would warrant more investigation.</p>
<p>General questions you might ask about your input reads include:</p>
<ul>
<li>How good is my read set?</li>
<li>Do I need to ask for a new sequencing run?  </li>
<li>Is it suitable for the analysis I need to do?</li>
</ul>
<p>For a fuller discussion of FastQC outputs and warnings, see the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC website link</a>, including the section on each of the output <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">reports</a>, and examples of <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">&ldquo;good&rdquo;</a> and <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">&ldquo;bad&rdquo;</a> Illumina data.</p>
<h2 id="what-next">What Next?</h2>
<p><a href="../spades/">Assemble the reads into a draft genome sequence</a>.</p>
<!--
- Trim reads with [Trimmomatic.](../trimmomatic/index.md)
-->

<!---
FIXME: include these?

- link to a fastqc protocol:
http://vlsci.github.io/lscc_docs/tutorials/assembly/assembly-protocol/#section-1-read-quality-control

- more detailed information:
https://docs.google.com/document/pub?id=16GwPmwYW7o_r-ZUgCu8-oSBBY1gC97TfTTinGDk98Ws
--->
            
          
        </div>
      </div>
    </div>
    <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
      <div class="sphinxsidebarwrapper">
        
  <h1 class="logo"><a href="../..">McGill SI 2018</a></h1>



  <p class="blurb">Tutorials for analysing bacterial omics data</p>

        
          



<h3>Table Of Contents</h3>

<nav>
  
  
    <ul>
    
      <li><a href="#learning-objectives">Learning Objectives</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#input-files">Input files</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#run-fastqc">Run FastQC</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#examine-output-files">Examine output files</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#what-next">What Next?</a></li>
      <ul>
    
  </ul>
    
  </ul>
  

  
</nav>
          
  <h3>Related Topics</h3>
  <ul>
    
      <li>Previous: <a href="../data-dna/" title="previous chapter">
        Training data set
      </a></li>
    
    
      <li>Next: <a href="../spades/" title="next chapter">
        Genome assembly
      </a></li>
    
  </ul>

        
        <div id="searchbox" style="display: none;" role="search">
  <h3>Quick search</h3>
  <form class="search" action="../../search.html" method="get">
    <input name="q" type="text">
    <input value="Go" type="submit">
  </form>
  <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
  </p>
</div>
<script type="text/javascript">
  document.getElementById("searchbox").style.display = "block";
</script>
      </div>
    </div>
    <div class="clearer"></div>
  </div>

  
    <div class="footer">
      
        &copy; &copy; The authors; 2016-
      
      
        |
        Powered by <a href="http://www.mkdocs.org">MkDocs 0.17.2</a>
        &amp; <a href="https://github.com/iamale/mkdocs-alabaster">Alabaster</a>
      
    </div>
  

  <!--
  MkDocs version      : 0.17.2
  Docs Build Date UTC : 2018-06-11 23:54:54
  -->
</body>
</html>