
<!DOCTYPE html>
<html>
<head>
  
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">

    <title>Genome assembly &mdash; McGill SI 2018</title>

    <link rel="shortcut icon" href="../../img/favicon.ico">
    <link rel="stylesheet" href="../../css/alabaster.css" type="text/css">
    <link rel="stylesheet" href="../../css/alabaster-overrides.css" type="text/css">

    
      <link href="../../css/sepsis.css" rel="stylesheet">
    

    
      <script src="../../search/require.js"></script>
    
      <script src="../../search/search.js"></script>
    

    

    <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9">

    
  
</head>
<body>

  <div class="document">
    <div class="documentwrapper">
      <div class="bodywrapper">
        <div class="body" role="main">
          
            <p><br></p>
<h1 id="assembly-using-spades">Assembly using Spades</h1>
<p>Keywords: de novo assembly, Spades, Galaxy, Microbial Genomics Virtual Lab</p>
<h2 id="background">Background</h2>
<p>Spades is one of a number of <em>de novo</em> assemblers that use short read sets as input (e.g. Illumina Reads), and the assembly method is based on de Bruijn graphs. For information about Spades see this <a href="http://bioinf.spbau.ru/spades">link</a>.</p>
<!---
A protocol for assembling with Velvet (another *de novo* assembler) is available [here](https://docs.google.com/document/d/1xs-TI5MejQARqo0pcocGlymsXldwJbJII890gnmjI0o/pub).
--->

<p>In this activity, we will perform a <em>de novo</em> assembly of a short read set using the Spades assembler. The output from Spades that we are interested in is a multiFASTA file that contains the draft genome sequence.</p>
<!--
We have a closed, annotated genome sequence for a closely related *wildtype* strain.
-->

<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<!--1. import data into Galaxy  
2. view the input read files -->

<ol>
<li>assemble the reads using Spades, and</li>
<li>examine the output assembly.</li>
</ol>
<!-- ## Import data into Galaxy
-  Go to the address of your galaxy server.

- [Remind me how to logon.](https://docs.google.com/document/d/1LAQvhIG8s-vv6T14bb8lGRkmoNha7E3bHf9kAgUwMs0/pub)
FIXME: note this contains the same galaxy address as above - change?


- Click on the <ss>Analyze Data</ss> menu at the top of the page.    
- Click on the <ss>History options</ss> button the ![history button](images/image02.png) on the top right of the history pane.
- Click <ss>Import from File</ss> (at the bottom of the list).  
- A new page will appear with a text box for the URL of the history to import.  
- Copy the following URL into the text box: <http://43.240.98.1/public/dieter/Galaxy-History-Colombiaworkshopstart.tar.gz>  
- Click <ss>Submit</ss>.  
- Galaxy will download the data files from the internet and will be available as an additional history (takes about one minute).  
- To view this new history, click the <ss>View all histories</ss> button![Histories button](images/image01.png) (top right of the history pane).  
  - If the history has finished downloading it will appear as <fn>imported from archive: Colombia_workshop_start</fn>
- Click on the ![Switch button](images/image06.png) button above the <fn>imported from archive:Colombia_workshop_start</fn> then the ![Done button](images/image05.png) button.
- You should now have four files in the history pane as follows:

![Files in history](images/image07.png)

--->

<h2 id="import-and-view-data">Import and view data</h2>
<!-- If you have completed the previous tutorial on [Quality Control](/modules/fastqc/index.md), you should already have the required files in your current Galaxy history. If not, see how to get them [here](/modules/galaxy/index.md).
-->

<p>See <a href="../data-dna/">here</a> for the link to import the Galaxy history for this tutorial, if you don&rsquo;t already have them in your history.</p>
<ul>
<li>The read set for today is from an imaginary <em>Staphylococcus aureus</em> bacterium with a miniature genome.</li>
<li>
<p>The whole genome shotgun method used to sequence our mutant strain read set was produced on an Illumina DNA sequencing instrument.</p>
</li>
<li>
<p>The files we need for assembly are the <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>.</p>
</li>
<li>
<p>(We don&rsquo;t need the reference genome sequences for this tutorial).</p>
</li>
<li>
<p>The reads are paired-end.</p>
</li>
<li>
<p>Each read is 150 bases long. <!--(before trimming)--></p>
</li>
<li>
<p>The number of bases sequenced is equivalent to 19x the genome sequence of the wildtype strain. (Read coverage 19x - rather low!).</p>
</li>
</ul>
<!--
- <fn>wildtype.fna</fn>: the reference genome sequence of the wildtype strain in fasta format (a header line, then the nucleotide sequence of the genome)

- <fn>wildtype.gff</fn>: the reference genome sequence of the wildtype strain in general feature format (a list of features - one feature per line, then the nucleotide sequence of the genome).

- <fn>wildtype.gbk</fn>: the reference genome sequence in genbank format.
--->

<ul>
<li>Click on the View Data button (the <img alt="Eye icon" src="./images/image04.png" />) next to one of the FASTQ sequence files.</li>
</ul>
<!--
- The gff file should look like this:
- Brief Discussion about the GFF format (FIXME: add)
![GFF format](images/image08.png)

## Evaluate the input reads

Questions you might ask about your input reads include:

- How good is my read set?
- Do I need to ask for a new sequencing run?  
- Is it suitable for the analysis I need to do?

We will evaluate the input reads using the FastQC tool.

- This runs a standard series of tests on your read set and returns a relatively easy-to-interpret report.
- We will use the FastQC tool in Galaxy to evaluate the quality of one of our FASTQ files.
- Go to <ss>Tools &rarr; NGS:Analysis &rarr; NGS: QC and Manipulation &rarr; FastQC</ss>
- Select <fn>mutant_R1.fastq</fn>
- <ss>Execute</ss>
- Once finished, examine the output called <fn>FastQC on data1:webpage</fn> (Hint:![Eye icon](./images/image04.png)). It has a summary at the top of
the page and a number of graphs.

Some of the important outputs of FastQC for our purposes are:

-   <ss>Basic Statistics: Sequence length</ss>: will be important in setting maximum k-mer size value for assembly
-   <ss>Basic Statistics: Encoding</ss>: Quality encoding type: important for quality trimming software
-   <ss>Basic Statistics: % GC</ss>: high GC organisms don’t tend to assemble well and may have an uneven read coverage distribution.
-   <ss>Basic Statistics: Total sequences</ss>: Total number of reads: gives you an idea of coverage.
-   <ss>Per base sequence quality</ss>: Dips in quality near the beginning, middle or end of the reads: determines possible trimming/cleanup methods and parameters and may indicate technical problems with the sequencing process/machine run.
-   <ss>Per base N content</ss>: Presence of large numbers of Ns in reads: may point to poor quality sequencing run. You would need to trim these reads to remove Ns.
-   <ss>Kmer content</ss>: Presence of highly recurring k-mers: may point to contamination of reads with barcodes or adapter sequences.

Although we have warnings for two outputs (per base sequence content; Kmer content), we can ignore these for now. For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of ["good"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and ["bad"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data. We won’t be doing anything to these data to clean it up as there isn’t much need. Therefore we will get on with the assembly!

-->

<h2 id="assemble-reads-with-spades">Assemble reads with Spades</h2>
<ul>
<li>We will perform a <em>de novo</em> assembly of the mutant FASTQ reads into long contiguous sequences (in FASTA format.)</li>
</ul>
<!---
- Spades produces both contigs and scaffolds.
Ask your demonstrator if you would like to know the difference between contigs and scaffolds.
--->

<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; spades</ss></li>
<li>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li><ss>Run only Assembly</ss>: <em>Yes</em> [the <em>Yes</em> button should be darker grey]</li>
<li><ss>Kmers to use separated by commas:</ss> <em>33,55,91</em>  [note: no spaces]  </li>
<li><ss>Coverage cutoff:</ss> <em>auto</em>  </li>
<li><ss>Files &rarr; Forward reads:</ss> <fn>mutant_R1.fastq</fn>  </li>
<li><ss>Files &rarr; Reverse reads:</ss> <fn>mutant_R2.fastq</fn>  </li>
</ul>
</li>
<li>
<p>Your tool interface should look like this:</p>
</li>
</ul>
<p><img alt="Spades interface" src="./images/image03.png" /></p>
<ul>
<li>Click <ss>Execute</ss></li>
</ul>
<h2 id="examine-the-output">Examine the output</h2>
<ul>
<li>Galaxy is now running Spades on the reads for you.</li>
<li>
<p>When it is finished, you will have five new files in your history.  </p>
<ul>
<li>two FASTA files of the resulting contigs and scaffolds</li>
<li>two files for statistics about these</li>
<li>the Spades logfile</li>
</ul>
</li>
</ul>
<p><img alt="spades output" src="./images/output_files.png" /></p>
<ul>
<li>Click on the View Data button <img alt="Eye icon" src="./images/image04.png" /> on each of the files.</li>
<li>Note that the short reads have been assembled into much longer contigs.</li>
<li>(However, in this case, the contigs have not been assembled into larger scaffolds.)</li>
<li>The stats files will give you the length of each of the contigs.</li>
</ul>
<p><img alt="spades output contigs" src="./images/contig_stats.png" /></p>
<!-- ## What next?

- [Annotate the genome using Prokka.](/modules/prokka/index.md)
-->
            
          
        </div>
      </div>
    </div>
    <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
      <div class="sphinxsidebarwrapper">
        
  <h1 class="logo"><a href="../..">McGill SI 2018</a></h1>



  <p class="blurb">Tutorials for analysing bacterial omics data</p>

        
          



<h3>Table Of Contents</h3>

<nav>
  
  
    <ul>
    
      <li><a href="#background">Background</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#learning-objectives">Learning objectives</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#import-and-view-data">Import and view data</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#assemble-reads-with-spades">Assemble reads with Spades</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#examine-the-output">Examine the output</a></li>
      <ul>
    
  </ul>
    
  </ul>
  

  
</nav>
          
  <h3>Related Topics</h3>
  <ul>
    
      <li>Previous: <a href="../fastqc/" title="previous chapter">
        Quality control
      </a></li>
    
    
      <li>Next: <a href="../prokka/" title="next chapter">
        Genome annotation
      </a></li>
    
  </ul>

        
        <div id="searchbox" style="display: none;" role="search">
  <h3>Quick search</h3>
  <form class="search" action="../../search.html" method="get">
    <input name="q" type="text">
    <input value="Go" type="submit">
  </form>
  <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
  </p>
</div>
<script type="text/javascript">
  document.getElementById("searchbox").style.display = "block";
</script>
      </div>
    </div>
    <div class="clearer"></div>
  </div>

  
    <div class="footer">
      
        &copy; &copy; The authors; 2016-
      
      
        |
        Powered by <a href="http://www.mkdocs.org">MkDocs 0.17.2</a>
        &amp; <a href="https://github.com/iamale/mkdocs-alabaster">Alabaster</a>
      
    </div>
  

  <!--
  MkDocs version      : 0.17.2
  Docs Build Date UTC : 2018-06-12 05:15:22
  -->
</body>
</html>