
<!DOCTYPE html>
<html>
<head>
  
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">

    <title>Variant finding &mdash; McGill SI 2018</title>

    <link rel="shortcut icon" href="../../img/favicon.ico">
    <link rel="stylesheet" href="../../css/alabaster.css" type="text/css">
    <link rel="stylesheet" href="../../css/alabaster-overrides.css" type="text/css">

    
      <link href="../../css/sepsis.css" rel="stylesheet">
    

    
      <script src="../../search/require.js"></script>
    
      <script src="../../search/search.js"></script>
    

    

    <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9">

    
  
</head>
<body>

  <div class="document">
    <div class="documentwrapper">
      <div class="bodywrapper">
        <div class="body" role="main">
          
            <p><br></p>
<h1 id="variant-calling-with-snippy">Variant calling with Snippy</h1>
<p>Keywords: variant calling, SNP, Snippy, JBrowse, Galaxy, Microbial Genomics Virtual Lab</p>
<h2 id="background">Background</h2>
<p>Variant calling is the process of identifying differences between two genome samples.
Usually differences are limited to single nucleotide polymorphisms (SNPs) and small insertions and deletions (indels). Larger structural variation such as inversions, duplications and large deletions are not typically covered by &ldquo;variant calling&rdquo;.</p>
<p>In this tutorial, we will use the tool &ldquo;Snippy&rdquo; (link to Snippy is <a href="https://github.com/tseemann/snippy">here</a>). Snippy uses a tool to align the reads to a reference genome, and another tool to decide (&ldquo;call&rdquo;) if the discrepancies are real variants.</p>
<h2 id="learning-objectives">Learning Objectives</h2>
<ol>
<li>Find variants between a reference genome and a set of reads</li>
<li>Visualise the SNP in context of the reads aligned to the genome</li>
<li>Determine the effect of those variants on genomic features</li>
<li>Understand if the SNP is potentially affecting the phenotype</li>
</ol>
<h2 id="prepare-reference">Prepare reference</h2>
<!-- We will import  history from Galaxy:

- In the menu options across the top, go to <ss>Shared Data</ss>.
- Click on <ss>Histories</ss>.
- A list of published histories should appear. Click on <fn>GCC 2016 small genome variants</fn>.
- Click on <ss>Import history</ss>.
- An option will appear to re-name the history. We don't need to rename it, so click <ss>Import</ss>.
- The history will now appear in your Current History pane, and the files are now ready to use in Galaxy analyses.
-->

<!-- We will use the same data that we used in the [Assembly with Spades tutorial.](../spades/index.md) This should still be in your current galaxy history. If not, re-import the data into a new history using the instructions in that tutorial.-->

<p>For variant calling, we need a reference genome that is of the same strain as the input sequence reads.</p>
<p>For this tutorial, our reference is the <fn>wildtype.gbk</fn> file and our reads are <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>.</p>
<p>If these files are not presently in your Galaxy history, import them from the <a href="../data-dna/">Training dataset page.</a></p>
<h2 id="call-variants-with-snippy">Call variants with Snippy</h2>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Variant Analysis &rarr; snippy</ss></li>
<li>For <ss>Reference type</ss> select <em>Genbank</em>.</li>
<li>Then for <ss>Reference Genbank</ss> choose the <fn>wildtype.gbk</fn> file.</li>
<li>For <ss>Single or Paired-end reads</ss> choose <em>Paired</em>.</li>
<li>Then choose the first set of reads, <fn>mutant_R1.fastq</fn> and second set of reads, <fn>mutant_R2.fastq</fn>.</li>
<li>For <ss>Cleanup the non-snp output files</ss> select <em>No</em>.</li>
</ul>
<p>Your tool interface should look like this:</p>
<p><img alt="Snippy interface" src="./images/interface.png" /></p>
<ul>
<li>Click <ss>Execute</ss>.</li>
</ul>
<h2 id="examine-snippy-output">Examine Snippy output</h2>
<!-- First, enable "Scratchbook" in Galaxy - this allows you to view several windows simultaneously. Click on the squares:

![scratchbook icon](images/scratchbook.png)

-->

<p>From Snippy, there are 10 output files in various formats.</p>
<ul>
<li>Go to the file called <fn>snippy on data XX, data XX and data XX table</fn> and click on the eye icon.</li>
<li>We can see a list of variants. Look in column 3 to see which types the variants are, such as a SNP or a deletion.</li>
<li>Look at the third variant called. This is a T&rarr;A mutation, causing a stop codon. Look at column 14: the product of this gene is a methicillin resistance protein. Methicillin is an antibiotic. What might be the result of such a mutation? <!--[add a hint/info box]--></li>
</ul>
<h2 id="view-snippy-output-in-jbrowse">View Snippy output in JBrowse</h2>
<ul>
<li>
<p>Go to <ss>Statistics and Visualisation &rarr; Graph/Display Data &rarr; JBrowse genome browser</ss>.</p>
</li>
<li>
<p>Under <ss>Reference genome to display</ss> choose <em>Use a genome from history</em>.</p>
</li>
<li>
<p>Under <ss>Select the reference genome</ss> choose <fn>wildtype.fna</fn>. This sequence will be the reference against which annotations are displayed.</p>
</li>
<li>
<p>For <ss>Produce a Standalone Instance</ss> select <em>Yes</em>.</p>
</li>
<li>
<p>For <ss>Genetic Code</ss> choose <em>11: The Bacterial, Archaeal and Plant Plastid Code</em>.</p>
</li>
<li>
<p>Under <ss>JBrowse-in-Galaxy Action</ss> choose <em>New JBrowse Instance</em>.</p>
</li>
<li>
<p>We will now set up three different tracks - these are datasets displayed underneath the reference sequence (which is displayed as nucleotides in FASTA format). We will choose to display the sequence reads (the .bam file), the variants found by snippy (the .gff file) and the annotated reference genome (the wildtype.gff)</p>
</li>
</ul>
<p><em>Track 1 - sequence reads</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss></li>
<li>For <ss>Track Cateogry</ss> name it &ldquo;sequence reads&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>BAM Pileups</em></li>
<li>For <ss>BAM Track Data</ss> select <fn>the snippy bam file</fn></li>
<li>For <ss>Autogenerate SNP Track</ss> select <em>Yes</em></li>
<li>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</li>
</ul>
<p><em>Track 2 - variants</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss> again</li>
<li>For <ss>Track Category</ss> name it &ldquo;variants&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>GFF/GFF3/BED/GBK Features</em></li>
<li>For <ss>Track Data</ss> select <fn>the snippy snps gff file</fn></li>
<li>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</li>
</ul>
<p><em>Track 3 - annotated reference</em></p>
<ul>
<li>Click <ss>Insert Track Group</ss> again</li>
<li>For <ss> Track Category</ss> name it &ldquo;annotated reference&rdquo;</li>
<li>Click <ss>Insert Annotation Track</ss></li>
<li>For <ss>Track Type</ss> choose <em>GFF/GFF3/BED/GBK Features</em></li>
<li>For <ss>Track Data</ss> select <fn>wildtype.gff</fn></li>
<li>Under <ss>JBrowse Track Type[Advanced]</ss> select <em>Canvas Features</em>.</li>
<li>Click on <ss>JBrowse Styling Options <Advanced]</ss></li>
<li>Under <ss>JBrowse style.label</ss> add in the word <em>product</em>.</li>
<li>Under <ss>JBrowse style.description</ss> add in the word <em>product</em>.</li>
<li>
<p>Under <ss>Track Visibility</ss> choose <em>On for new users</em>.</p>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
<li>
<p>A new file will be created, called <fn>JBrowse on data XX and data XX - Complete</fn>. Click on the eye icon next to the file name. The JBrowse window will appear in the centre Galaxy panel.</p>
</li>
<li>
<p>On the left, tick boxes to display the tracks</p>
</li>
<li>
<p>Use the minus button to zoom out to see:</p>
<ul>
<li>sequence reads and their coverage (the grey graph)</li>
</ul>
</li>
<li>
<p>Use the plus button to zoom in to see:</p>
<ul>
<li>probable real variants (a whole column of snps)</li>
<li>probable errors (single one here and there)</li>
</ul>
</li>
</ul>
<p><img alt="JBrowse screenshot" src="./images/jbrowse1.png" /></p>
<ul>
<li>In the coordinates box, type in <em>47299</em> and then <ss>Go</ss> to see the position of the SNP discussed above.<ul>
<li>the correct codon at this position is TGT, coding for the amino acid Cysteine, in the middle row of the amino acid translations.</li>
<li>the mutation of T &rarr; A turns this triplet into TGA, a stop codon.</li>
</ul>
</li>
</ul>
<p><img alt="JBrowse screenshot" src="./images/jbrowse2.png" />    </p>
            
          
        </div>
      </div>
    </div>
    <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
      <div class="sphinxsidebarwrapper">
        
  <h1 class="logo"><a href="../..">McGill SI 2018</a></h1>



  <p class="blurb">Tutorials for analysing bacterial omics data</p>

        
          



<h3>Table Of Contents</h3>

<nav>
  
  
    <ul>
    
      <li><a href="#background">Background</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#learning-objectives">Learning Objectives</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#prepare-reference">Prepare reference</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#call-variants-with-snippy">Call variants with Snippy</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#examine-snippy-output">Examine Snippy output</a></li>
      <ul>
    
  </ul>
    
      <li><a href="#view-snippy-output-in-jbrowse">View Snippy output in JBrowse</a></li>
      <ul>
    
  </ul>
    
  </ul>
  

  
</nav>
          
  <h3>Related Topics</h3>
  <ul>
    
      <li>Previous: <a href="../prokka/" title="previous chapter">
        Genome annotation
      </a></li>
    
    
      <li>Next: <a href="../abricate/" title="next chapter">
        Finding antibiotic-resistance genes
      </a></li>
    
  </ul>

        
        <div id="searchbox" style="display: none;" role="search">
  <h3>Quick search</h3>
  <form class="search" action="../../search.html" method="get">
    <input name="q" type="text">
    <input value="Go" type="submit">
  </form>
  <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
  </p>
</div>
<script type="text/javascript">
  document.getElementById("searchbox").style.display = "block";
</script>
      </div>
    </div>
    <div class="clearer"></div>
  </div>

  
    <div class="footer">
      
        &copy; &copy; The authors; 2016-
      
      
        |
        Powered by <a href="http://www.mkdocs.org">MkDocs 0.17.2</a>
        &amp; <a href="https://github.com/iamale/mkdocs-alabaster">Alabaster</a>
      
    </div>
  

  <!--
  MkDocs version      : 0.17.2
  Docs Build Date UTC : 2018-06-13 05:08:10
  -->
</body>
</html>